src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -151,63 +151,54 @@ private ImmutableArray<TypeSymbol> GetDeclaredConstraintTypes()
             }
 
             var moduleSymbol = containingType.ContainingPEModule;
-
-            EntityHandle[] constraints;
+            var metadataReader = moduleSymbol.Module.MetadataReader;
 
             try
             {
-                constraints = moduleSymbol.Module.GetGenericParamConstraintsOrThrow(_handle);
-            }
-            catch (BadImageFormatException)
-            {
-                constraints = null;
-                Interlocked.CompareExchange(ref _lazyBoundsErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
-            }
-
-            if (constraints != null && constraints.Length > 0)
-            {
-                var symbolsBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
-                MetadataDecoder tokenDecoder;
+                var constraints = metadataReader.GetGenericParameter(_handle).GetConstraints();
```

Comment:
>var [](start = 16, length = 3)

Please follow the guideline about not using ```var``` when type isn't explicitly spelled in the initializer.